File: models/ambulance/AmbulanceMaster.js
// models/ambulance/AmbulanceMaster.js
import mongoose from "mongoose";

const AmbulanceMasterSchema = new mongoose.Schema({
  hospitalId: { type: mongoose.Schema.Types.ObjectId, ref: "Hospital", required: true, index: true },
  branchId: { type: mongoose.Schema.Types.ObjectId, ref: "Branch", required: true, index: true },
  vehicleNumber: { type: String, required: true, unique: true },
  type: { type: String, enum: ["Basic", "Advanced", "ICU"], default: "Basic" },
  driverId: { type: mongoose.Schema.Types.ObjectId, ref: "User" }, // Driver
  contactNumber: String,
  isActive: { type: Boolean, default: true },
}, { timestamps: true });

export default mongoose.model("AmbulanceMaster", AmbulanceMasterSchema);

File: jstojv/src/main/java/com/samrat/jstojv/entity/AmbulanceMaster.java
package com.samrat.jstojv.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "ambulance_masters")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AmbulanceMaster extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "hospital_id", nullable = false)
    private Hospital hospital;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "branch_id", nullable = false)
    private Branch branch;

    @Column(name = "vehicle_number", nullable = false, unique = true)
    private String vehicleNumber;

    @Enumerated(EnumType.STRING)
    private Type type = Type.Basic;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "driver_id")
    private User driver;

    @Column(name = "contact_number")
    private String contactNumber;

    @Column(name = "is_active")
    private boolean isActive = true;

    public enum Type {
        Basic, Advanced, ICU
    }
}
